pkg/actuators/machineset/controller.go

```diff
@@ -90,14 +90,15 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return result, err
 }
 
-func reconcile(machineSet *machinev1.MachineSet) (ctrl.Result, error) {
+func reconcile(logger logr.Logger, machineSet *machinev1.MachineSet) (ctrl.Result, error) {
 	providerConfig, err := getProviderConfig(*machineSet)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to get providerConfig: %v", err)
```

Comment:
Maybe we should do the same here as well?  If someone updates the provider config later to be valid, we'll see a new event to reconcile.

Reply:
just so i understand, you mean we should log an error and return `ctrl.Result{}, nil` ?